### DISCUSSION TITLE
Add a new disk.

### DISCUSSION THREAD

**Discussion by bsouza-site** [netbox-community/netbox#10658] [Ideas]
Hello Everyone!

In the inventory I'm doing, I need to add the disks of a VM or a host.

In some cases VM\host has two disks, **is it possible to add the field "Disk Space 2"?**

![1](https://user-images.githubusercontent.com/47142683/195943326-25b204fa-fa0b-455b-baa8-ded1e4b1d85f.PNG)

**Comment by candlerb** [comment]
No, not in these standard fields.  I think it's expected that you will put the total of the two disks in here.  After all, this is just a high-level summary; there is no further detail available around storage pools, storage classes etc.

If this were a physical device then you'd use "Inventory Items", but they don't exist in Netbox for VMs.

I think if you want to document (or push-configure) virtual machines with this level of detail, then you'll need to go with [custom fields](https://docs.netbox.dev/en/stable/customization/custom-fields/), or else develop a plugin which documents "VM resources" that can be attached to a VM.

**Reply by ziggekatten** [reply]
We use config context to hold disk info through our integration with VMware. Sure, readability is not the best, but since we need that info machine readable anyway for charging and stuff, it works pretty well.

**Comment by jeremystretch** [comment]
See also FR #8356, which proposes the introduction of an analog to inventory items for virtual machines.

**Comment by pdenessen** [comment]
We are running in this as well as we want to launch VMs in VMware via ansible based on the info and parameters set in Netbox. Therefore a couple of fields are missing and I think best would be to add Inventory for VMs as wekk as stated in above FR.

**Comment by Lord-ZuzurNC** [comment]
I think this idea can be closed as Virtual Machines now integrates Virtual Disks in an almost perfect way (but I'll create another idea for that thing it miss IMO).
![image](https://github.com/netbox-community/netbox/assets/90102166/d047ce73-c393-4b13-8d17-792e98991809)
![image](https://github.com/netbox-community/netbox/assets/90102166/70eeb768-4716-4f9a-93d2-67324f1beb5d)
